compiler/qsc/src/bin/qsc.rs

```diff
@@ -24,6 +27,28 @@ use std::{
     string::String,
 };
 
+#[derive(clap::ValueEnum, Clone, Debug, Default, PartialEq)]
+pub enum QirProfile {
+    /// This is the default profile, which allows all operations.
+    #[default]
+    Unrestricted,
+    /// This profile restricts the set of operations to those that are supported by the Base profile.
+    Base,
+    /// This profile restricts the set of operations to those that are supported by the AdaptiveRI profile.
+    AdaptiveRI,
+}
```

Comment:
I know not everyone is a fan of repeated names, but I'd much rather this be called `Profile`. It's an exact copy of the other `Profile` enum, and almost-the-same-but-different names for the same concept confuse me.